File: code_eval/java/moderate/Challenge16.java
package moderate;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;

/**
 * Challenge 16 : Number of ones
 * Created by cschalk on 3/21/16.
 */
public class Challenge16 {

    public static void main (String[] args) throws Exception {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            numberOfOnes(line);
        }
    }

    private static void numberOfOnes(String input) {
        String binary = Integer.toBinaryString(Integer.parseInt(input));
        int count = 0;
        for (char c : binary.toCharArray()) {
            if (c == '1') {
                count++;
            }
        }
        System.out.println(count);
    }
}

File: code_eval/cpp/moderate/challenge_16.cpp
// Created by cschalk on 3/30/16.
// Challenge 16 : Number of ones
#include <iostream>
#include <fstream>
#include <bitset>

int main(int argc, char ** argv) {
    std::ifstream input {argv[1]};
    int test;

    while (input >> test) {
        std::string binary = std::bitset<64>(test).to_string();
        int count = 0;
        for (char c : binary) {
            if (c == '1') {
                count++;
            }
        }
        std::cout << count << std::endl;
    }

    return 0;
}
